[Bug]: gradle build issues in windows

### Description

This issue contains a list of gradle issues present when building ballerina-lang on Windows.

1. `:bir-spec:kaitai`  task failure
Windows build fails for `:bir-spec:kaitai` task with the following exception,
```
* What went wrong:
Execution failed for task ':bir-spec:kaitai'.
> name.valery1707.kaitai.KaitaiException: Fail to generate Java files
```

2. `:jballerina-tools:generateDocs` task failure
If we exclude the above task and run the build again, following exception is thrown,
```
* What went wrong:
Execution failed for task ':jballerina-tools:generateDocs'.
> Process 'command 'C:\src\jdk-17.0.8.7-hotspot\bin\java.exe'' finished with non-zero exit value 1
```

3. `./gradlew clean` task failure
This is mentioned in https://github.com/ballerina-platform/ballerina-lang/issues/42782.
When running the `./gradlew clean` after a build, following exception is thrown,
```
Execution failed for task ':clean'.
> java.io.IOException: Unable to delete directory 'C:\Users\Ballerina_CodeBase\BallerinaLangClones\ballerina-lang\build'
    Failed to delete some children. This might happen because a process has files open or has its working directory set in the target directory.
    - C:\Users\Ballerina_CodeBase\BallerinaLangClones\ballerina-lang\build\tmp\.cache\expanded\expanded.lock
    - C:\Users\Ballerina_CodeBase\BallerinaLangClones\ballerina-lang\build\tmp\.cache\expanded
    - C:\Users\Ballerina_CodeBase\BallerinaLangClones\ballerina-lang\build\tmp\.cache
    - C:\Users\Ballerina_CodeBase\BallerinaLangClones\ballerina-lang\build\tmp
```

As a workaround for failure 1 and 2, use the following command,
```
./gradlew build -x check x :jballerina-tools:generateDocs -x :bir-spec:kaitai
```

### Steps to Reproduce

To reproduce above mentioned errors run the following commands in `ballerina-lang` directory.

1. `:bir-spec:kaitai`  task failure
```
./gradlew build -x check 
```

2. `:jballerina-tools:generateDocs` task failure

```
./gradlew build -x check -x :bir-spec:kaitai
```

3. `./gradlew clean` task failure
Run the following command after attempting a build,
```
./gradlew clean
```

### Affected Version(s)

Ballerina 2201.10.0 (Swan Lake Update 10)

### OS, DB, other environment details and versions

OS: Windows 10

### Related area

-> CICD

### Related issue(s) (optional)

https://github.com/ballerina-platform/ballerina-lang/issues/42782
https://github.com/ballerina-platform/ballerina-lang/issues/41090

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

PR #42782 just got merged. For me, `./gradlew build -x check` is now working on Windows. 
Also directly running `./gradlew :bir-spec:kaitai` works for me.
If I execute `./gradlew :jballerina-tools:generateDocs` I get following error:
```
> Task :docerina-gradle-plugin:createJavadoc
E:\Programming\open-source\ballerina-lang\misc\ballerina-gradle-plugins\docerina-gradle-plugin\src\main\java\org\ballerinalang\plugin\gradle\doc\DocerinaGen.java:33: warning: no comment
    public static void main(String[] args) {
                       ^
error: warnings found and -Werror specified
1 error
1 warning
```

@Shadow-Devil, If you try running the gradle task with the `--info` flag, it will print the full stacktrace.

```
./gradlew :jballerina-tools:generateDocs --info
```

You will get the following error,
```
Exception in thread "main" java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
        at org.ballerinalang.docgen.docs.BallerinaDocGenerator.<clinit>(BallerinaDocGenerator.java:70)
        at org.ballerinalang.plugin.gradle.doc.DocerinaGen.main(DocerinaGen.java:34)
Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
        ... 2 more
```
This error seems to be the actual cause for `./gradlew :jballerina-tools:generateDocs` not working. The reason for this might be due to `slf4j` dependency not being properly provided. Hope this is helpful. I will also try to find the root cause and update here if I find it.

--- a/misc/ballerina-gradle-plugins/docerina-gradle-plugin/src/main/java/org/ballerinalang/plugin/gradle/doc/DocerinaGen.java
+++ b/misc/ballerina-gradle-plugins/docerina-gradle-plugin/src/main/java/org/ballerinalang/plugin/gradle/doc/DocerinaGen.java
@@ -30,6 +30,10 @@ public class DocerinaGen {
 
     private static final PrintStream out = System.out;
 
+    /**
+     * Main method to generate Ballerina API docs.
+     * @param args command line arguments
+     */
     public static void main(String[] args) {
         BallerinaDocGenerator.mergeApiDocs(Path.of(args[0]));
     }

